.github/workflows/pull_request.yml

```diff
@@ -37,12 +37,14 @@ jobs:
       run: |
         UMBREL_OS_VERSION="dev"
         IMAGE_NAME="umbrel-os-dev"
+        REPO="$(curl -s https://api.github.com/repos/getumbrel/umbrel/pulls/${PR} | jq .head.repo.full_name | sed 's/"//g')"
         UMBREL_REPO="$(curl -s https://api.github.com/repos/getumbrel/umbrel/pulls/${PR} | jq .head.repo.clone_url | sed 's/"//g')"
-        UMBREL_BRANCH="$(curl -s https://api.github.com/repos/getumbrel/umbrel/pulls/${PR} | jq .head.ref | sed 's/"//g')"
-        echo "::set-env name=UMBREL_OS_VERSION::$UMBREL_OS_VERSION"
-        echo "::set-env name=IMAGE_NAME::$IMAGE_NAME"
-        echo "::set-env name=UMBREL_REPO::$UMBREL_REPO"
-        echo "::set-env name=UMBREL_BRANCH::$UMBREL_BRANCH"
+        BRANCH="$(curl -s https://api.github.com/repos/getumbrel/umbrel/pulls/${PR} | jq .head.ref | sed 's/"//g')"
+        UMBREL_BRANCH="$(curl https://api.github.com/repos/$REPO/branches/$BRANCH | jq .commit.sha | sed 's/"//g')"
```

Comment:
This is vulnerable to the same bug as #272, but with a reduced time interval.

If you push two commits soon after each other, the first action could start after the second commit, meaning when you query the GitHub API for the tip of the branch, you get the same commit for both builds.

Instead of querying the API you can just use the preset GitHub Actions environment variables.

https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables

`GITHUB_SHA` should be the commit SHA that triggered the workflow which is exactly what we want.